addons/product/models/product_product.py

```diff
@@ -425,6 +435,221 @@ def _onchange_uom_id(self):
             }
         }
 
+    @api.depends('product_template_attribute_value_ids')
+    def _export_attribute_values(self):
+        for p in self:
+            p.import_attribute_values = ','.join(sorted(
+                f'{v.attribute_line_id.attribute_id.name}:{v.product_attribute_value_id.name}'
+                for v in p.product_template_attribute_value_ids
+            ))
+
+    def _load_records_write(self, values):
+        import_attribute_values = values.pop('import_attribute_values', '')
+        if not import_attribute_values:
+            return super()._load_records_write(values)
+
+        attribute_values = import_attribute_values.split(',')
+        existing_values = self.import_attribute_values.split(',')
+        wrong = len(attribute_values) != len(existing_values)
+        if not wrong:
+            for value in attribute_values:
+                attribute_name, value_name = value.split(':', 1)
+                wrong = f'{attribute_name.strip()}:{value_name.strip()}' not in existing_values
```

Comment:
`wrong` is wrongly determined.  Consider two attributes, where the first one is wrong but the second one is correct.  After the loop, `wrong` is `False`, which is wrong 😄 
```suggestion
        wrong = len(attribute_values) != len(existing_values) or any(
            len(item := value.split(':', 1)) != 2 or f'{item[0].strip()}:{item[1].strip()}' not in existing_values
            for value in attribute_values
        )
```